### ISSUE TITLE
SpeechAsyncClient calls 'next' on async request generator

### ISSUE THREAD

**Issue by 0xfadeef** [googleapis/python-speech#110]
#### Environment details

  - OS type and version: OS X `10.11.6`
  - Python version: `3.8.6`
  - pip version: `20.2.1`
  - `google-cloud-speech` version: `2.0.1`

#### Steps to reproduce
I'm trying to employ `SpeechAsyncClient` to perform asynchronous stream recognition. Here's a snippet from the code I came up with:
```python
speech_client = SpeechAsyncClient.from_service_account_json(...)
config_request = ...

async def _generate_requests(voice_stream: 'async_generator'):
    yield config_request
    async for chunk in voice_stream:
        request = StreamingRecognizeRequest()
        request.audio_content = chunk
        yield request

async def recognize_stream(voice_stream):
    recognize_agen = speech_client.streaming_recognize(_generate_requests(voice_stream))

    transcripts = [
        response.results[0].alternatives[0].transcript
        async for response in recognize_agen if response.results[0].is_final]
    ...

```
Here `voice_stream` is an async generator which yields chunks of audio to recognize. The problem is that `await`ing on `recognize_stream(voice_stream)` raises unexpected exception:

#### Stack trace
```
ERROR:grpc._channel:Exception iterating requests!
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/grpc/_channel.py", line 195, in consume_request_iterator
    request = next(request_iterator)
TypeError: 'async_generator' object is not an iterator
```

It seems like `SpeechAsyncClient` calls `next` on a request generator instead of treating it like an _async_ generator, which I reckon it's supposed to do. I couldn't find any documentation on this API and can only assume the expected behavior, but it surely looks like a bug in implementation. Or am I missing something?

**Comment by b-loved-dreamer** [comment]
Hi @0xfadeef, we have a working sample that I believe can help you achieve your goal. Please, check this [sample](https://github.com/googleapis/python-speech/blob/master/samples/microphone/transcribe_streaming_infinite.py).

**Comment by 0xfadeef** [comment]
Hello @b-loved-dreamer! I looked through the example you provided, but I'm not sure how it can help in my situation. AFAICT it uses synchronous `SpeechClient` and the problem I'm having is that I couldn't make `SpeechAsyncClient` to work.

More specifically, for some reason, `streaming_recognize()` method, which according to the [documentation](https://googleapis.dev/python/speech/latest/speech_v1/services.html#google.cloud.speech_v1.services.speech.SpeechAsyncClient.streaming_recognize) expects `AsyncIterator` as its first argument, still treats it like a plain iterator and tries to call `next()` on it.

I tried to `pip install --upgrade` all Google related packages (like `google-api-core`), but still no luck. Thanks anyways! :)

**Comment by 0xfadeef** [comment]
So, it seems like this issue is caused by `SpeechAsyncClient.from_service_account_json` being merely a reference to its `SpeechClient` counterpart. Calling this method creates an instance of `SpeechClient` instead of `SpeechAsyncClient`.

Since the problem has been fixed in commit 3bed0b43c75be649e29475240c7f486fc9cd63dc, now merged into mainline, I close this issue.
